src/containers/pages/AssigmentMapWrapper/index.tsx

```diff
@@ -32,18 +49,93 @@ export interface AssignmentMapWrapperProps {
   currentChildren: TreeNode[];
   serviceClass: typeof OpenSRPService;
   fetchJurisdictionsActionCreator: typeof fetchJurisdictions;
+  fetchUpdatedCurrentParentActionCreator: typeof fetchUpdatedCurrentParent;
   getJurisdictionsFeatures: FeatureCollection;
+  autoSelectNodesActionCreator: ActionCreator<AutoSelectNodesAction>;
+  selectNodeCreator: ActionCreator<SelectNodeAction>;
+  deselectNodeCreator: ActionCreator<DeselectNodeAction>;
+  currentParentNode: TreeNode | undefined;
 }
 
 const defaultProps = {
+  autoSelectNodesActionCreator: autoSelectNodes,
   currentChildren: [],
   currentParentId: undefined,
+  currentParentNode: undefined,
+  deselectNodeCreator: deselectNode,
   fetchJurisdictionsActionCreator: fetchJurisdictions,
+  fetchUpdatedCurrentParentActionCreator: fetchUpdatedCurrentParent,
   getJurisdictionsFeatures: undefined,
   rootJurisdictionId: '',
+  selectNodeCreator: selectNode,
   serviceClass: OpenSRPService,
 };
 
+/**
+ * Handler to get current parent id from clicked jurisdiction on map
+ * @param props - AssignmentMapWrapper component props
+ */
+
+export const onJurisdictionClick = (props: any, setMapParent: any) => {
+  function handleJurisdictionClick(_: Map, e: EventData) {
+    // Destructure event data
+    const { point, target, originalEvent } = e;
+    const {
+      fetchUpdatedCurrentParentActionCreator,
+      currentParentNode,
+      rootJurisdictionId,
+      selectNodeCreator,
+      deselectNodeCreator,
+    } = props;
+    // grab underlying features from map
+    const features: any = target.queryRenderedFeatures(point) as Dictionary[];
+    // dont drill down map if current jurisdiction is leafNode
+    if (!features.length) {
+      return false;
+    }
+    if (features[0]) {
```

Comment:
What's the significance of 0 here?  Can we avoid using a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) like this? I'd prefer either a named constant or a map.

Reply:
It seems that `features` is an array of something but we are only using the first element.  Why?  Would it make sense to structure it different if we only need one element?

Reply:
The reason for this is sometimes when features on map are being queried it would return an array of duplicate feature objects